forms/models/form.js

```diff
@@ -371,6 +382,22 @@ define(function (require) {
     /* eslint-disable no-unused-vars */ //stop eslint compaining about options and errorList not being used.
     setErrors: function(errorList, options){
       var elementsCollection = this.get('elements');
+      var subForms = this.getSubforms();
+
+      _.each(subForms, function(subForm, name){
+        var subFormErrorList = _.omit(errorList[name], 'errors');
+
+        _.each(subFormErrorList, function(errors, formName){
+          var forms = subForms[name].where({name: formName});
+          _.each(errors, function(fieldError, formIndex){
+            var form = subForms[name].at(formIndex);
+            if ( form ){
+              form.setErrors(fieldError);
+            }
+          });
+        });
+      });
+
```

Comment:
for loops will always be faster, but I find `each` to be more readable. Its debatable which is better :) Also, when the sub form errors are on forms that have sequential ids, PHP will encode it as an array and not as an object. using `_.each` normalises this so we dont have to worry about it. 

I can get rid of the `var  forms = ... ` part though. No idea why i left that!

I would of *loved* to not have to triple nest it, but the data structure of the errors and of forms didnt offer much help.

First off you cant tell an error is for a subform without checking the actual element for a type property. The error just doesnt have that information, and the only reliable way was to check if the field name of the error returned an element of type 'subForm'

The other main thing is that this needs to be done on every level of subforms. The errors are not reported as a flat structure (they cant be, because the property names are not unique!). You have no choice but to traverse it down, making sure to go down the *form* the correct index of the *subform element*. This function should only go down the branches that have an error, not every subform that exists.

When there are no subforms, it just delegates setting the errors to the element collection. 

subforms make my head hurt. i'll remove the unnecessary `var forms` line